### ISSUE TITLE
kubelet Failed create pod sandbox got wiretype 0, want 2

### ISSUE THREAD

**Issue by yu-yang2** [kubernetes/frakti#307]
I use [https://github.com/kubernetes/frakti/blob/master/docs/deploy.md](url) this guide to deploy k8s with frakti, everything works fine except I run a deploy with `kubectl run nginx --image=nginx`
```
[root@yangyu-pc plugins]# kubectl describe po nginx-8586cf59-gvxpv 
Name:           nginx-8586cf59-gvxpv
Namespace:      default
Node:           yangyu-pc/192.168.0.100
Start Time:     Mon, 12 Mar 2018 05:13:58 -0400
Labels:         pod-template-hash=41427915
                run=nginx
Annotations:    <none>
Status:         Pending
IP:             
Controlled By:  ReplicaSet/nginx-8586cf59
Containers:
  nginx:
    Container ID:   
    Image:          nginx
    Image ID:       
    Port:           <none>
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-nx6ls (ro)
Conditions:
  Type           Status
  Initialized    True 
  Ready          False 
  PodScheduled   True 
Volumes:
  default-token-nx6ls:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-nx6ls
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                  Age               From                Message
  ----     ------                  ----              ----                -------
  Normal   Scheduled               4m                default-scheduler   Successfully assigned nginx-8586cf59-gvxpv to yangyu-pc
  Normal   SuccessfulMountVolume   4m                kubelet, yangyu-pc  MountVolume.SetUp succeeded for volume "default-token-nx6ls"
  Warning  FailedCreatePodSandBox  0s (x20 over 4m)  kubelet, yangyu-pc  Failed create pod sandbox.
```
and kubelet logs
```
Mar 12 05:15:08 yangyu-pc kubelet[4735]: E0312 05:15:08.122961    4735 kuberuntime_manager.go:647] createPodSandbox for pod "nginx-8586cf59-gvxpv_default(b271e42e-25d5-11e8-8842-74d435e40eea)" failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2
Mar 12 05:15:08 yangyu-pc kubelet[4735]: E0312 05:15:08.122996    4735 pod_workers.go:186] Error syncing pod b271e42e-25d5-11e8-8842-74d435e40eea ("nginx-8586cf59-gvxpv_default(b271e42e-25d5-11e8-8842-74d435e40eea)"), skipping: failed to "CreatePodSandbox" for "nginx-8586cf59-gvxpv_default(b271e42e-25d5-11e8-8842-74d435e40eea)" with CreatePodSandboxError: "CreatePodSandbox for pod \"nginx-8586cf59-gvxpv_default(b271e42e-25d5-11e8-8842-74d435e40eea)\" failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2"
Mar 12 05:15:18 yangyu-pc kubelet[4735]: E0312 05:15:18.131876    4735 remote_runtime.go:92] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2
Mar 12 05:15:18 yangyu-pc kubelet[4735]: E0312 05:15:18.131913    4735 kuberuntime_sandbox.go:54] CreatePodSandbox for pod "kube-dns-6f4fd4bdf-5jl8n_kube-system(7aa7cb6f-25d5-11e8-8842-74d435e40eea)" failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2
Mar 12 05:15:18 yangyu-pc kubelet[4735]: E0312 05:15:18.131921    4735 kuberuntime_manager.go:647] createPodSandbox for pod "kube-dns-6f4fd4bdf-5jl8n_kube-system(7aa7cb6f-25d5-11e8-8842-74d435e40eea)" failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2
Mar 12 05:15:18 yangyu-pc kubelet[4735]: E0312 05:15:18.131960    4735 pod_workers.go:186] Error syncing pod 7aa7cb6f-25d5-11e8-8842-74d435e40eea ("kube-dns-6f4fd4bdf-5jl8n_kube-system(7aa7cb6f-25d5-11e8-8842-74d435e40eea)"), skipping: failed to "CreatePodSandbox" for "kube-dns-6f4fd4bdf-5jl8n_kube-system(7aa7cb6f-25d5-11e8-8842-74d435e40eea)" with CreatePodSandboxError: "CreatePodSandbox for pod \"kube-dns-6f4fd4bdf-5jl8n_kube-system(7aa7cb6f-25d5-11e8-8842-74d435e40eea)\" failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2"
Mar 12 05:15:20 yangyu-pc kubelet[4735]: E0312 05:15:20.131007    4735 remote_runtime.go:92] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2
Mar 12 05:15:20 yangyu-pc kubelet[4735]: E0312 05:15:20.131042    4735 kuberuntime_sandbox.go:54] CreatePodSandbox for pod "nginx-8586cf59-gvxpv_default(b271e42e-25d5-11e8-8842-74d435e40eea)" failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2
Mar 12 05:15:20 yangyu-pc kubelet[4735]: E0312 05:15:20.131053    4735 kuberuntime_manager.go:647] createPodSandbox for pod "nginx-8586cf59-gvxpv_default(b271e42e-25d5-11e8-8842-74d435e40eea)" failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2
Mar 12 05:15:20 yangyu-pc kubelet[4735]: E0312 05:15:20.131116    4735 pod_workers.go:186] Error syncing pod b271e42e-25d5-11e8-8842-74d435e40eea ("nginx-8586cf59-gvxpv_default(b271e42e-25d5-11e8-8842-74d435e40eea)"), skipping: failed to "CreatePodSandbox" for "nginx-8586cf59-gvxpv_default(b271e42e-25d5-11e8-8842-74d435e40eea)" with CreatePodSandboxError: "CreatePodSandbox for pod \"nginx-8586cf59-gvxpv_default(b271e42e-25d5-11e8-8842-74d435e40eea)\" failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Gateway: got wiretype 0, want 2"
```

**Comment by gavinmy15** [comment]
I follow the same tutorial and get the similar problem.

frakti.log
failed: rpc error: code = Unknown desc = proto: bad wiretype for field types.UserInterface.Mtu: got wiretype 2, want 0

**Comment by liyanyanli** [comment]
frakti has updated [types.proto](https://github.com/kubernetes/frakti/blob/master/pkg/hyper/types/types.proto#L434)(hyperd) to the latest version(in the master branch), so using the released frakti version such as v1.9.1 can avoid this problem.

**Comment by yu-yang2** [comment]
@liyanyanli thanks, problem solved, but  `frakti --version` shows version still 1.9, I build it from release v1.9.1.
```
[root@yangyu-pc kubernetes]# frakti --version
F0313 22:05:23.455201   31194 frakti.go:74] frakti version: 1.9
```
